modules/util/util.go

```diff
@@ -25,6 +27,27 @@ const (
 	OptionalBoolFalse
 )
 
+var matchComplexities = map[string]regexp.Regexp{}
+var matchComplexityOnce sync.Once
+
+// CheckPasswordComplexity return True if password is Complexity
+func CheckPasswordComplexity(pwd string) bool {
+	matchComplexityOnce.Do(func() {
+		for key, val := range setting.PasswordComplexity {
```

Comment:
This is unnecessarily complex. Can you revert to the original implementation with just one config value?

Reply:
Using the one-line checkout option with regexp causes additional complication. Checking in this case will depend on the sequence of indication of the group of characters.
example,
For option 
[a-z]+[A-Z]+[0-9_]+[^A-Za-z0-9_]+

, password "qqWw9*"  will be valid, and password "QiiW*3" meeting the requirements will be erroneous.

If we take into account all the options for 4 groups of characters, then the number of options in the expression will reach to 16 :
([a-z]+[A-Z]+[0-9_]+[^A-Za-z0-9_]+)|([A-Z]+[a-z]+[0-9_]+[^A-Za-z0-9_]+)|...

 Maybe i'm wrong and there is a simpler version of using a regular expression